src/mqt/yaqs/core/data_structures/state.py

```
@@ -0,0 +1,323 @@
+    during evolution (:attr:`representation`). Materialization happens at construction;
+    pass the ``State`` to :func:`~mqt.yaqs.simulator.run` (including in parameter loops).
+
+    - **Presets** — ``State(L, initial="zeros")``; default ``representation="mps"`` (TJM).
+      For MCWF or Lindblad, set ``representation="vector"`` or ``"density_matrix"``.
+    - **Manual data** — exactly one of ``tensors``, ``vector``, or ``density_matrix``; the
+      representation is inferred (do not pass ``representation=``).
+
+    Circuit simulation requires ``representation="mps"`` (checked in ``run``).
+    """
+
+    def __init__(
```

Ruff (0.15.12)

[warning] Too many branches (17 > 12)

(PLR0912)